chore: mark Import/Export classes as internal

ennasus4sun opened the pull request:
this PR is a proposal for that RFC: https://github.com/shopware/shopware/discussions/5982

ennasus4sun committed:
chore: mark Import/Export classes as internal

ennasus4sun commented on tests/unit/Core/Content/ImportExport/Message/ImportExportHandlerTest.php:
```diff
@@ -85,6 +86,27 @@ public function testImportExportHandlerDispatchesMessage(string $activity, strin
         static::assertSame($logEntity->getActivity(), $importExportMessage->getActivity());
     }
 
+    /**
```
FYI: its not new code. I just move the dataprovider closer to the test method which uses it.

mitelg commented on tests/unit/Core/Content/ImportExport/Event/ImportExportAfterProcessFinishedEventTest.php:
```diff
@@ -0,0 +1,34 @@
+ * @internal
+ */
+#[Package('fundamentals@after-sales')]
+class ImportExportAfterProcessFinishedEventTest extends TestCase
```
imo it is not reasonable to test event classes, as they do not contain any logic. just put the `@codeCoverageIgnore` annotation on the event and that's it :slightly_smiling_face:

ennasus4sun replied:
Yes, I have already thought about it... Thank you, Telgi

ennasus4sun committed:
Merge branch 'trunk' of github.com:shopware/shopware into next-40446/make-import-export-internal

ennasus4sun committed:
chore: adjustments regarding phpstan an event test

MalteJanz commented on src/Core/Content/ImportExport/Event/ImportExportAfterProcessFinishedEvent.php:
```diff
@@ -0,0 +1,40 @@
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * @internal
```
I guess the event itself shouldn't be internal right? Because we want people to use it / subscribe to it. Maybe we could make the constructor internal to allow future changes there, like adding more data.

MalteJanz commented on src/Core/Content/ImportExport/Strategy/Import/BatchImportStrategy.php:
```diff
@@ -75,6 +75,7 @@ public function commit(Config $config, Progress $progress, Context $context): Im
                 $result = $this->repository->upsert($records, $context);
             }
 
+            // @deprecated tag:v6.7.0 - remove the whole loop with no replacement
```
Things like this can also already be prepared by checking the major feature flag with an if, which we can remove next week with all the other deprecations.

Benefit of that approach: If you activate the feature flag now, you shouldn't get any deprecation warnings because that event class should already not be used anymore

MalteJanz commented on src/Core/Content/ImportExport/Strategy/Import/OneByOneImportStrategy.php:
```diff
@@ -49,6 +49,7 @@ public function import(
                 $result = $this->repository->upsert([$record], $context);
             }
 
+            // @deprecated tag:v6.7.0 - remove this event with no replacement
```
see above. Another benefit: extensions can already test with the major flag if they still work

MalteJanz commented:
One more thing: this change should include a changelog announcing the deprecations

ennasus4sun committed:
chore: remove triggerDeprication

from ImportExportAfterImportRecordEvent constructor which causes failing unit tests

ennasus4sun committed:
chore: add changelog

and prepare codeparts for v6.7.0.0

ennasus4sun committed:
chore: prepare tests for v6.7.0.0

ennasus4sun committed:
Merge branch 'trunk' of github.com:shopware/shopware into next-40446/make-import-export-internal

mitelg commented on changelog/_unreleased/2025-01-29-mark-import-export-internal.md:
```diff
@@ -0,0 +1,31 @@
+---
+title: mark Import/Export internal
```
```suggestion
title: Mark parts of the Import/Export functionality as internal
```

mitelg commented on changelog/_unreleased/2025-01-29-mark-import-export-internal.md:
```diff
@@ -0,0 +1,31 @@
+issue: NEXT-40446
+---
+# Core
+* Added `Shopware\Core\Content\ImportExport\Event\ImportExportAfterProcessFinishedEvent.php`
```
please use the class reference and not the file name and add a small sentence about its purpose

mitelg commented on changelog/_unreleased/2025-01-29-mark-import-export-internal.md:
```diff
@@ -0,0 +1,31 @@
+---
+# Core
+* Added `Shopware\Core\Content\ImportExport\Event\ImportExportAfterProcessFinishedEvent.php`
+* Deprecated `Shopware\Core\Content\ImportExport\Event\ImportExportAfterImportRecordEvent.php` which will be removed in 6.7.0.0 without replacement
```
also here and on the other occurrences here. do not reference the file, but he full class name instead

mitelg commented on changelog/_unreleased/2025-01-29-mark-import-export-internal.md:
```diff
@@ -0,0 +1,31 @@
+  * `Shopware\Core\Content\ImportExport\Processing\Pipe\PipeFactory.php`
+___
+# Next Major Version Changes
+## These classes are now marked as internal
```
this will be merged with the upgrade guide and will become a headline in there. so the headline of your changes would be something like 

```suggestion
## Changes to the import/export functionality
```

Then you can write something like: The following classes are now marked as internal: 
just as normal sentence

mitelg commented on changelog/_unreleased/2025-01-29-mark-import-export-internal.md:
```diff
@@ -0,0 +1,31 @@
+* `Shopware\Core\Content\ImportExport\Processing\Pipe\KeyMappingPipe.php`
+* `Shopware\Core\Content\ImportExport\Processing\Pipe\PipeFactory.php`
+
+## This class is now removed without replacement
```
instead of having this as a headline, you can write a sentence, that this class was removed

mitelg commented on changelog/_unreleased/2025-01-29-mark-import-export-internal.md:
```diff
@@ -0,0 +1,31 @@
+## This class is now removed without replacement
+* `Shopware\Core\Content\ImportExport\Event\ImportExportAfterImportRecordEvent.php`
+
+## This method is removed without replacement
```
also here, remove the headline and just add a sentence about what was done

ennasus4sun committed:
chore: adjust changelog

ennasus4sun committed:
Merge branch 'trunk' of github.com:shopware/shopware into next-40446/make-import-export-internal

ennasus4sun committed:
Merge branch 'trunk' of github.com:shopware/shopware into next-40446/make-import-export-internal
